### ISSUE TITLE
Do not merge lines and destroy formatting in JSDoc for @example tag

### ISSUE THREAD

**Issue by BITespresso** [20Tauri/DoxyDoxygen#127]
In JSDoc the **@example** tag is followed by some lines of code (see: [http://usejsdoc.org/tags-example.html](url)). Therefore, DoxyDoxygen should not merge those lines into a single paragraph without line breaks after **Alt-Q**.

Before:

```js
/**
 * Tries to resolve a promise multiple times if it is rejected. The passed function will be
 * called repeatedly until the promise returned by the function is resolved or the maximum
 * number of attempts is reached.
 *
 * Returns a new promise that is fulfilled with the fulfillment value of the promise
 * returned by the function, or is rejected with the same reason as the promise returned by
 * the function in the last attempt.
 *
 * Subsequent calls of the function will be deferred by a delay specified in milliseconds.
 *
 * @param      {Function}         fn      The function returning a promise.
 * @param      {number}           times   The maximum number of attempts.
 * @param      {number}           delay   The delay.
 * @return     {BIT.SDS.Promise}  A new promise.
 *
 * @example
 * var trySomething = function() {
 *   var resolver = function(resolve, reject) {
 *     ... // Try to resolve promise
 *   };
 *   return new BIT.SDS.Promise(resolver);
 * };
 * BIT.SDS.Promise.retry(trySomething, 5, 5000)
 *   .then(...)
 *   .catch(...);
 */
```

After Alt-Q:

```js
/**
 * Tries to resolve a promise multiple times if it is rejected. The passed function will be called
 * repeatedly until the promise returned by the function is resolved or the maximum number of
 * attempts is reached.
 *
 * Returns a new promise that is fulfilled with the fulfillment value of the promise returned by the
 * function, or is rejected with the same reason as the promise returned by the function in the last
 * attempt.
 *
 * Subsequent calls of the function will be deferred by a delay specified in milliseconds.
 *
 * @param      {Function}         fn      The function returning a promise.
 * @param      {number}           times   The maximum number of attempts.
 * @param      {number}           delay   The delay.
 * @return     {BIT.SDS.Promise}  A new promise.
 *
 * @example    var trySomething = function() { var resolver = function(resolve, reject) { ... // Try
 *             to resolve promise }; return new BIT.SDS.Promise(resolver); };
 *             BIT.SDS.Promise.retry(trySomething, 5, 5000) .then(...) .catch(...);
 */
```

This will unfortunately mess up the formatting of the code.

**Comment by 20Tauri** [comment]
Thanks for reporting,

I have just a 0.69.1 version (evolution branch only) that include this fix.

**Comment by BITespresso** [comment]
Great! Thanks!

**Comment by BITespresso** [comment]
I was using the version 0.69.1 very successfully but just discovered an issue with it: There might be several **@example** tags in a single JSDoc comment block. See:

```js
/**
 * Resets all application windows to their default sizes and to positions determined by the
 * specified rectangle. The algorithm used ensures that each window has a position that depends
 * entirely on the specified rectangle, regardless of which applications are installed or which
 * DSM version is used.
 *
 * Note 1: Open application windows will not change their size and position. You must manually
 * close and reopen the windows to see the effects of the reset. Do not move or resize an
 * already open application window, as this immediately sets the restore size and position again
 * to the current window size and position.
 *
 * Note 2: The Synology CMS (Central Management System) application (`SYNO.SDS.CMS.Application`)
 * does not read the stored window size and position due to a bug in DSM. To ensure that this
 * window has its designated position, the window will be opened each time this method is called
 * and set to the designated position.
 *
 * @param      {BIT.SDS.Rectangle=}  rectangle  The rectangle.
 *
 * @example
 * BIT.SDS.WindowUtil.cascadeOverlapAllWindows({x: 160, y: 139, width: 1640, height: 830});
 *
 * @example
 * BIT.SDS.WindowUtil.cascadeOverlapAllWindows();
 */
```

The issue with 0.69.1 is that those **two** examples get merged to just **one** example after Alt-Q:

```js
/**
 * Resets all application windows to their default sizes and to positions determined by the
 * specified rectangle. The algorithm used ensures that each window has a position that depends
 * entirely on the specified rectangle, regardless of which applications are installed or which DSM
 * version is used.
 *
 * Note 1: Open application windows will not change their size and position. You must manually close
 * and reopen the windows to see the effects of the reset. Do not move or resize an already open
 * application window, as this immediately sets the restore size and position again to the current
 * window size and position.
 *
 * Note 2: The Synology CMS (Central Management System) application (`SYNO.SDS.CMS.Application`)
 * does not read the stored window size and position due to a bug in DSM. To ensure that this window
 * has its designated position, the window will be opened each time this method is called and set to
 * the designated position.
 *
 * @param      {BIT.SDS.Rectangle=}  rectangle  The rectangle.
 *
 * @example
 * BIT.SDS.WindowUtil.cascadeOverlapAllWindows({x: 160, y: 139, width: 1640, height: 830});
 *
 * BIT.SDS.WindowUtil.cascadeOverlapAllWindows();
 */
```

This results in just **one** example in the final JSDoc, which is not intended.

**Comment by 20Tauri** [comment]
Thanks again for reporting. Just fixed in 0.69.2
